great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -316,12 +322,28 @@ class SQLDatasource(Datasource):
     connection_string: str
     assets: Dict[str, TableAsset] = {}
 
-    @property
-    def execution_engine_type(self) -> Type[ExecutionEngine]:
-        """Returns the default execution engine type."""
-        from great_expectations.execution_engine import SqlAlchemyExecutionEngine
+    # non-field private attrs
+    _engine: Engine = pydantic.PrivateAttr()
 
-        return SqlAlchemyExecutionEngine
+    @property
+    def engine(self) -> Engine:
+        return self._engine
+
+    class Config:
+        arbitrary_types_allowed = True
+        extra = pydantic.Extra.forbid
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._engine = create_engine(self.connection_string)
```

Comment:
I think this should be defined on the parent `Datasource`
And we can use `<DATASOURCE>._excluded_eng_args` to know which arguments to pass.

Maybe the concrete `DataAsset` or `DataSource` should define a `_create_engine()` method?? 🤔

Reply:
Only `SqlDatasource` needs a SQL Alchemy `engine` so it would seem to be an anti-pattern to promote this.

Reply:
Only `SqlDatasource` needs a `SQLAlchemy` engine as returned by `sqlalchemy.engine.create_engine` but everything needs an "ExectuionEngine".